### PR TITLE
[ff-142] new enemies (Demo)

### PR DESCRIPTION
**Pull request by blueUserRed**

### CODE REVIEW

File: core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt [hunk 1]
```diff
@@ -22,94 +36,132 @@ sealed class EnemyAction(val showProbability: Float, protected val enemy: Enemy)
...
-                ).string
-            }
-            includeLater(
-                { controller.confirmationPopupTimeline(text) },
-                { true }
-            )
-            action {
+        override fun create(controller: GameController, scale: Double): EnemyAction {
+            val cardAmount = controller.cardHand.cards.size
+            val amountToDestroy = (1..maxCards).random().coerceAtMost(cardAmount - 1)
```
Comment (GreenUserBlue) [inline core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt:74]: why not just irgnore the coerce and put it into the 1..maxCards part? (should be possible i think)

File: core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt [hunk 2]
```diff
@@ -22,94 +36,132 @@ sealed class EnemyAction(val showProbability: Float, protected val enemy: Enemy)
...
-    ) : EnemyAction(showProbability, enemy) {
-
-        override fun getTimeline(controller: GameController, scale: Double): Timeline = Timeline.timeline {
-            include(controller.enemyAttackTimeline(damage.scale(scale * scaleFactor).random()))
+        hasUnlikelyPredicates: Boolean,
+        private val iconHandle: ResourceHandle?,
+    ) : EnemyActionPrototype(showProbability, enemy, hasUnlikelyPredicates) {
+
+        override fun create(controller: GameController, scale: Double): EnemyAction {
+            val damage = damage.scale(scale * scaleFactor).random()
```
[the remark this answers is not in the document]
Comment (blueUserRed) [reply core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt:54]: what is wrong with the random function? In my opinion its easier to read than the Random object

File: core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt (hunk 1 above)
Comment (blueUserRed) [reply core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt:74]: I could change it to `(1..maxCards.coerceAtMost(cardAmount - 1)).random()`, but in my opinion that is even harder to read

File: core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (GreenUserBlue) [reply core/src/com/fourinachamber/fortyfive/game/enemy/EnemyAction.kt:54]: i meant using a gerneral Random object 
like 
var rnd: Random = Random(100)
and then u use it like

... .random(rnd)

**Review by GreenUserBlue — APPROVED** [review]

**Pull request merged**